core/trino-main/src/main/java/io/trino/operator/scalar/MathFunctions.java

```diff
@@ -1398,4 +1398,36 @@ private static Double mapL2Norm(Block map)
 
         return Math.sqrt(norm);
     }
+
+    @Description("Black-Scholes equation for a Euro vanilla option")
+    @ScalarFunction("bs_european_option")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double bsEuropeanOption(
+            @SqlType(StandardTypes.DOUBLE) double stockPrice,
+            @SqlType(StandardTypes.DOUBLE) double strikePrice,
+            @SqlType(StandardTypes.DOUBLE) double timeToExpiration,
+            @SqlType(StandardTypes.DOUBLE) double riskFreeRate,
+            @SqlType(StandardTypes.DOUBLE) double volatility,
+            @SqlType("varchar(x)") String optionType)
```

Comment:
@Praveen2112 can you confirm @SqlType `varchar(x)` and `String` are the best variable types to use here?